### PR TITLE
Fix locality label missing

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Supersude https://github.com/istio/istio/pull/42263

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -613,7 +613,7 @@ func setTopologyLabels(proxy *model.Proxy) {
 		localityStr = proxy.ServiceInstances[0].Endpoint.Locality.Label
 	} else {
 		// If no service instances(this maybe common for a pure client), respect LocalityLabel
-		localityStr = proxy.Metadata.Labels[model.LocalityLabel]
+		localityStr = proxy.Labels[model.LocalityLabel]
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads.go:616]: This is because we set proxy.Labels with `SetWorkloadLabels`, but used  Metadata.Labels here

**Comment by hzxuzhonghu** [comment]
/retest

@ramaraochavali

File: pkg/bootstrap/config_test.go [hunk 2]
```diff
@@ -92,6 +93,16 @@ func TestGetNodeMetaData(t *testing.T) {
 	t.Setenv(IstioMetaPrefix+"OWNER", inputOwner)
 	t.Setenv(IstioMetaPrefix+"WORKLOAD_NAME", inputWorkloadName)
 
+	// prepare a pod label file
+	os.MkdirAll("./etc/istio/pod/", os.ModePerm)
```
Comment (howardjohn) [inline pkg/bootstrap/config_test.go:97]: can use t.TempDir and os.chdir

File: pkg/bootstrap/config.go [hunk 3]
```diff
@@ -625,6 +625,10 @@ func GetNodeMetaData(options MetadataOptions) (*model.Node, error) {
 		l = options.Platform.Locality()
 	} else {
 		localityString := model.GetLocalityLabelOrDefault(meta.Labels[model.LocalityLabel], "")
+		if localityString != "" {
+			// replace "." with "/"
```
Comment (howardjohn) [inline pkg/bootstrap/config.go:629]: doesn't GetLocalityLabelOrDefault already do this?

**Comment by linsun** [comment]
@hzxuzhonghu curious why this doesn't need a release note?

File: pkg/bootstrap/config_test.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/bootstrap/config_test.go:99]: done

Comment (howardjohn) [reply pkg/bootstrap/config_test.go:101]: doesn't seem pushed?

**Review by howardjohn — APPROVED** [review]
hold since I don't think this has a local fix, but lgtm

File: pkg/bootstrap/config_test.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/bootstrap/config_test.go:101]: pushed, it create a tmp dire and changed dir, this is to create `/etc/istio/pod/` under the team dir.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**